GitUI/UserControls/RevisionGrid/ArtificialCommitChangeCount.cs

```diff
@@ -39,23 +44,28 @@ public sealed class ArtificialCommitChangeCount
         /// Any change in any category.
         /// </summary>
         public bool HasChanges
-            => (Changed?.Count ?? 0) > 0
+            => IsUpdating
+               && ((Changed?.Count ?? 0) > 0
                || (New?.Count ?? 0) > 0
                || (Deleted?.Count ?? 0) > 0
                || (SubmodulesChanged?.Count ?? 0) > 0
-               || (SubmodulesDirty?.Count ?? 0) > 0;
+               || (SubmodulesDirty?.Count ?? 0) > 0);
 
         /// <summary>
         /// Update the change count.
         /// </summary>
         /// <param name="items">Git items.</param>
-        public void Update(IReadOnlyList<GitItemStatus> items)
+        public void Update(IReadOnlyList<GitItemStatus>? items)
         {
-            Changed = items.Where(item => !item.IsNew && !item.IsDeleted && !item.IsSubmodule).ToList();
-            New = items.Where(item => item.IsNew && !item.IsSubmodule).ToList();
-            Deleted = items.Where(item => item.IsDeleted && !item.IsSubmodule).ToList();
-            SubmodulesChanged = items.Where(item => item.IsSubmodule && item.IsChanged).ToList();
-            SubmodulesDirty = items.Where(item => item.IsSubmodule && item.IsDirty).ToList();
+            IsUpdating = items is not null;
+            if (IsUpdating)
+            {
+                Changed = items.Where(item => !item.IsNew && !item.IsDeleted && !item.IsSubmodule).ToList();
+                New = items.Where(item => item.IsNew && !item.IsSubmodule).ToList();
+                Deleted = items.Where(item => item.IsDeleted && !item.IsSubmodule).ToList();
+                SubmodulesChanged = items.Where(item => item.IsSubmodule && item.IsChanged).ToList();
+                SubmodulesDirty = items.Where(item => item.IsSubmodule && item.IsDirty).ToList();
+            }
         }
```

Comment:
The name `IsUpdating` does not make any sense here - at least to me. With the value inverted, it would make more sense but would still not tell the full truth: We are not _updating_ it while modal forms are displayed.
Please untie the knots in my brain and rename it back to `Valid` or `Known`.

Reply:
@RussKie can I switch back to Valid?

Reply:
Changed to DataValid